src/app/campaign/flight/flight.component.ts

```diff
@@ -87,32 +88,25 @@ export class FlightComponent implements OnInit {
     });
   }
 
-  // emits updates when reactive form fields change
-  formStatusChanged(flight: Flight, changed?: boolean) {
-    this.flightUpdate.emit({
-      flight: { ...flight, totalGoal: this.flight.totalGoal, dailyMinimum: this.flight.dailyMinimum },
-      changed,
-      valid: this.flightForm.valid
-    });
+  isFlightFormChanged(flight: Flight) {
+    return (
+      isNameChanged(flight, this.flight) ||
+      isStartAtChanged(flight, this.flight) ||
+      isEndAtChanged(flight, this.flight) ||
+      isInventoryChanged(flight, this.flight) ||
+      isZonesChanged(flight, this.flight)
+    );
   }
 
-  onDateRangeChange({ startAt, endAt }: { startAt?: Date; endAt?: Date }) {
-    // these dates pickers are not actually connected to the reactive form
-    // so in order for the flightForm to be valid on emit, these values must be set on the form
-    if (startAt) {
-      this.flightForm.get('startAt').setValue(startAt, { emitEvent: false });
-    }
-    if (endAt) {
-      this.flightForm.get('endAt').setValue(endAt, { emitEvent: false });
+  // emits updates when reactive form fields change
+  formStatusChanged(flight: Flight) {
+    if (this.isFlightFormChanged(flight)) {
```

Comment:
Even the boolean guard on updateFlightForm was failing me, had to move to checking that the fields had changed.
I'll be looking to do the refactor on this with a parent form and embedded form components next.